diff --git a/.gitignore b/.gitignore
index bfd27ae..6f1ef91 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,17 @@
-/target/
+*.class
 */target/*
+target/*
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
+*.jar
+*.war
+*.ear
+*.orig
+*.rej
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+hs_err_pid*
+/crud-archetype/src/main/resources/archetype-resources/client/gen/
\ No newline at end of file
diff --git a/README.md b/README.md
index d727db5..3d04b67 100644
--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ There is simple archetype to create a basic application:
 mvn archetype:generate 
 	-DarchetypeGroupId=com.dukescript.archetype
 	-DarchetypeArtifactId=knockout4j-archetype 
-	-DarchetypeVersion=0.8
+	-DarchetypeVersion=0.9
 ```
 
 There is another, more complex archetype to generate skeleton of a CRUD application that shows how you can do 
@@ -23,7 +23,7 @@ code on both ends:
 mvn archetype:generate 
 	-DarchetypeGroupId=com.dukescript.archetype
 	-DarchetypeArtifactId=crud4j-archetype 
-	-DarchetypeVersion=0.8
+	-DarchetypeVersion=0.9
 ```
 
 Both archetypes can generate subprojects for each of the supported platforms. Currently we support 
diff --git a/crud-archetype-test/src/test/java/com/dukescript/archetype/crud/test/VerifyArchetypeIT.java b/crud-archetype-test/src/test/java/com/dukescript/archetype/crud/test/VerifyArchetypeIT.java
index b84905a..3d32ac0 100644
--- a/crud-archetype-test/src/test/java/com/dukescript/archetype/crud/test/VerifyArchetypeIT.java
+++ b/crud-archetype-test/src/test/java/com/dukescript/archetype/crud/test/VerifyArchetypeIT.java
@@ -25,9 +25,12 @@ package com.dukescript.archetype.crud.test;
 
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
 import java.io.FileReader;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.URL;
 import java.nio.file.FileVisitResult;
 import java.nio.file.FileVisitor;
@@ -38,6 +41,7 @@ import java.util.Properties;
 import java.util.jar.JarFile;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
 import javax.xml.XMLConstants;
 import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.parsers.ParserConfigurationException;
@@ -106,11 +110,13 @@ public class VerifyArchetypeIT {
                 fail("No pre-compilaton:\n" + l);
             }
         }
-        
-        v.verifyTextInLog("fxcompile/o-a-test/client/target/o-a-test-1.0-SNAPSHOT-javafx.zip");
-        
+        verifyFileInLog(v, "fxcompile/o-a-test/client/target/o-a-test-1.0-SNAPSHOT-javafx.zip");
         v.assertFilePresent("server/nbactions.xml");
     }
+
+    private void verifyFileInLog(Verifier v, final String t) throws VerificationException {
+        v.verifyTextInLog(t.replace('/', File.separatorChar));
+    }
     
     @Test public void iosProjectCompiles() throws Exception {
         final File dir = new File("target/tests/icompile/").getAbsoluteFile();
@@ -141,7 +147,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("install");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("icompile/o-b-test/client/target/o-b-test-1.0-SNAPSHOT-javafx.zip");
+        verifyFileInLog(v, "icompile/o-b-test/client/target/o-b-test-1.0-SNAPSHOT-javafx.zip");
         
         File client = new File(created, "client-ios");
         assertTrue(client.isDirectory(), "Subproject dir found: " + client);
@@ -248,7 +254,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("package");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("noicompile/m-n-test/client/target/m-n-test-1.0-SNAPSHOT-javafx.zip");
+        verifyFileInLog(v, "noicompile/m-n-test/client/target/m-n-test-1.0-SNAPSHOT-javafx.zip");
         
         File nbactions = new File(created, "nbactions.xml");
         assertTrue(nbactions.isFile(), "Actions file is in there");
@@ -263,6 +269,12 @@ public class VerifyArchetypeIT {
         assertTrue(created.isDirectory(), "Project created");
         assertTrue(new File(created, "pom.xml").isFile(), "Pom file is in there");
 
+        File index = new File(new File(new File(new File(new File(
+            created, "src"), "main"), "webapp"), "pages"), "index.html");
+        assertTrue(index.exists(), "HTML page found " + index);
+        File bin = new File(index.getParentFile(), "index.bin");
+        writeBinary(bin);
+        
         File and = new File(new File(dir, "d-l-test"), "android-test");
         assertTrue(and.isDirectory(), "Project created");
         assertTrue(new File(and, "pom.xml").isFile(), "Pom file is in there");
@@ -301,6 +313,9 @@ public class VerifyArchetypeIT {
         
         JarFile jf = new JarFile(apk);
         assertNotNull(jf.getEntry("assets/pages/index.html"), "index.html is included in " + apk);
+        ZipEntry indexBin = jf.getEntry("assets/pages/index.bin");
+        assertNotNull(indexBin, "binary file found in " + apk);
+        assertBinary(jf.getInputStream(indexBin));
         jf.close();
     }
 
@@ -325,7 +340,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("verify");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("wandroidcmp/w-d-test/client/target/w-d-test-1.0-SNAPSHOT-javafx.zip");
+        verifyFileInLog(v, "wandroidcmp/w-d-test/client/target/w-d-test-1.0-SNAPSHOT-javafx.zip");
     }
 
     @Test
@@ -336,14 +351,17 @@ public class VerifyArchetypeIT {
         File created = new File(new File(dir, "b-p-test"), "client");
         assertTrue(created.isDirectory(), "Project created");
         assertTrue(new File(created, "pom.xml").isFile(), "Pom file is in there");
-        
-        File web = new File(new File(dir, "b-p-test"), "test-web");
-        assertTrue(web.isDirectory(), "Project created");
-        assertTrue(new File(web, "pom.xml").isFile(), "Pom file is in there");
-        
+
         File main = new File(new File(created, "src"), "main");
         File pages = new File(new File(main, "webapp"), "pages");
         File index = new File(pages, "index.html");
+        assertTrue(index.exists(), "HTML page found " + index);
+        File bin = new File(index.getParentFile(), "index.bin");
+        writeBinary(bin);
+                
+        File web = new File(new File(dir, "b-p-test"), "test-web");
+        assertTrue(web.isDirectory(), "Project created");
+        assertTrue(new File(web, "pom.xml").isFile(), "Pom file is in there");
         
         String indexContent = Files.readFile(index);
         assertTrue(indexContent.contains("src=\"bck2brwsr.js\""), "There should be bck2brwsr.js reference in " + index);
@@ -360,7 +378,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("package");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("b2bcmp/b-p-test/test-web/target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
+        verifyFileInLog(v, "b2bcmp/b-p-test/test-web/target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
         
         v.assertFileNotPresent("target/res/drawable-hdpi/ic_launcher.png");
         v.assertFileNotPresent("target/res/drawable-mdpi/ic_launcher.png");
@@ -372,6 +390,10 @@ public class VerifyArchetypeIT {
         v.assertFilePresent("target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
         v.assertFilePresent("target/b-p-test.js");
         v.assertFilePresent("target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr/public_html/index.html");
+        v.assertFilePresent("target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr/public_html/index.bin");
+        File indexBin = new File(new File(new File(new File(web, "target"), "b-p-test-web-1.0-SNAPSHOT-bck2brwsr"), "public_html"), "index.bin");
+        assertTrue(indexBin.exists(), "index.bin really exists");
+        assertBinary(new FileInputStream(indexBin));
 
         File nbactions = new File(web, "nbactions.xml");
         assertTrue(nbactions.isFile(), "Actions file is in there");
@@ -419,7 +441,7 @@ public class VerifyArchetypeIT {
             v.executeGoal("package");
 
             v.verifyErrorFreeLog();
-            v.verifyTextInLog("BandN/b-n-test/for-web/target/b-n-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
+            verifyFileInLog(v, "BandN/b-n-test/for-web/target/b-n-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
 
             v.assertFileNotPresent("target/res/drawable-hdpi/ic_launcher.png");
             v.assertFileNotPresent("target/res/drawable-mdpi/ic_launcher.png");
@@ -565,4 +587,20 @@ public class VerifyArchetypeIT {
         Document dom = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(u.openStream());
         return xp.evaluate(dom);
     }    
+    
+    static void writeBinary(File out) throws IOException {
+        FileOutputStream fos = new FileOutputStream(out);
+        for (int i = 0; i < 256; i++) {
+            fos.write(i);
+        }
+        fos.close();
+    }
+    
+    static void assertBinary(InputStream is) throws IOException {
+        for (int i = 0; i < 256; i++) {
+            int b = is.read();
+            assertEquals((byte)b, (byte)i, i + "th byte of the stream should be equal");
+        }
+        is.close();
+    }
 }
diff --git a/crud-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml b/crud-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml
index f42dd74..12773f6 100644
--- a/crud-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml
+++ b/crud-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml
@@ -22,6 +22,8 @@
         </goals>
         <properties>
             <skipTests>true</skipTests>
+            <android.run.debug>${jpda.attach.port}</android.run.debug>
+            <jpda.attach>true</jpda.attach>
         </properties>
     </action>
     <action>
diff --git a/crud-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml b/crud-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml
index 56a713a..5df7721 100644
--- a/crud-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml
+++ b/crud-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml
@@ -18,7 +18,7 @@
 
     <properties>
         <platform.version>4.1.1.4</platform.version>
-        <android.plugin.version>4.1.0</android.plugin.version>
+        <android.plugin.version>4.2.0</android.plugin.version>
         <android.platform.version>19</android.platform.version>
         <debug>false</debug>
     </properties>
diff --git a/crud-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml b/crud-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml
index 2ff682b..a445c5d 100644
--- a/crud-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml
+++ b/crud-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml
@@ -18,6 +18,9 @@
         </goals>
         <properties>
             <skipTests>true</skipTests>
+            <robovm.debug>clientmode</robovm.debug>
+            <robovm.debugPort>${jpda.address}</robovm.debugPort>
+            <jpda.listen>true</jpda.listen>
         </properties>
     </action>
     <action>
diff --git a/crud-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml b/crud-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml
index 013fba7..47e8a3b 100644
--- a/crud-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml
+++ b/crud-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml
@@ -145,16 +145,13 @@
             <plugin>
                 <groupId>org.robovm</groupId>
                 <artifactId>robovm-maven-plugin</artifactId>
-                <version>1.0.0-beta-03</version>
+                <version>${robovm.version}</version>
                 <configuration>
                     <config>
                         <mainClass>\${project.mainclass}</mainClass>
                         <forceLinkClasses>
                             <pattern>java.util.logging.ConsoleHandler</pattern>
                             <pattern>java.util.logging.SimpleFormatter</pattern>
-                            <pattern>org.netbeans.html.ko4j.KO4J</pattern>
-                            <pattern>org.netbeans.html.sound.impl.BrowserAudioEnv</pattern>
-                            <pattern>com.dukescript.presenters.iOS</pattern>
                         </forceLinkClasses>
                         <executableName>\${artifactId}</executableName>
                         <resources>
diff --git a/crud-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml b/crud-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml
index 9114753..2427596 100644
--- a/crud-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml
+++ b/crud-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml
@@ -37,7 +37,6 @@
       <fileSet>
           <directory>target/web/pages</directory>
           <outputDirectory>/</outputDirectory>
-          <filtered>true</filtered>
       </fileSet>
         <fileSet>
             <directory>${project.build.directory}</directory>
diff --git a/crud-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml b/crud-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml
index 871719b..dd4b2f3 100644
--- a/crud-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml
+++ b/crud-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml
@@ -36,22 +36,7 @@
       <fileSet>
           <directory>src/main/webapp/pages</directory>
           <outputDirectory>/</outputDirectory>
-          <filtered>true</filtered>
-          <excludes>
-              <exclude>**/*.png</exclude>
-              <exclude>**/*.jpg</exclude>
-              <exclude>**/*.gif</exclude>
-          </excludes>
-      </fileSet>
-      <fileSet>
-          <directory>src/main/webapp/pages</directory>
-          <outputDirectory>/</outputDirectory>
           <filtered>false</filtered>
-          <includes>
-              <include>**/*.png</include>
-              <include>**/*.jpg</include>
-              <include>**/*.gif</include>
-          </includes>
       </fileSet>
   </fileSets>
 </assembly>
diff --git a/ko-archetype-test/src/test/java/com/dukescript/archetype/ko/test/VerifyArchetypeIT.java b/ko-archetype-test/src/test/java/com/dukescript/archetype/ko/test/VerifyArchetypeIT.java
index c708a96..4148206 100644
--- a/ko-archetype-test/src/test/java/com/dukescript/archetype/ko/test/VerifyArchetypeIT.java
+++ b/ko-archetype-test/src/test/java/com/dukescript/archetype/ko/test/VerifyArchetypeIT.java
@@ -25,9 +25,12 @@ package com.dukescript.archetype.ko.test;
 
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
 import java.io.FileReader;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.URL;
 import java.nio.file.FileVisitResult;
 import java.nio.file.FileVisitor;
@@ -38,6 +41,7 @@ import java.util.Properties;
 import java.util.jar.JarFile;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
 import javax.xml.XMLConstants;
 import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.parsers.ParserConfigurationException;
@@ -106,8 +110,12 @@ public class VerifyArchetypeIT {
                 fail("No pre-compilaton:\n" + l);
             }
         }
-        
-        v.verifyTextInLog("fxcompile/o-a-test/client/target/o-a-test-1.0-SNAPSHOT-javafx.zip");
+        final String t = "fxcompile/o-a-test/client/target/o-a-test-1.0-SNAPSHOT-javafx.zip";
+        verifyFileInLog(v, t);
+    }
+
+    private void verifyFileInLog(Verifier v, final String t) throws VerificationException {
+        v.verifyTextInLog(t.replace('/', File.separatorChar));
     }
     
     @Test public void iosProjectCompiles() throws Exception {
@@ -139,7 +147,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("install");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("icompile/o-b-test/client/target/o-b-test-1.0-SNAPSHOT-javafx.zip");
+        verifyFileInLog(v, "icompile/o-b-test/client/target/o-b-test-1.0-SNAPSHOT-javafx.zip");
         
         File client = new File(created, "client-ios");
         assertTrue(client.isDirectory(), "Subproject dir found: " + client);
@@ -246,7 +254,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("package");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("noicompile/m-n-test/client/target/m-n-test-1.0-SNAPSHOT-javafx.zip");
+        verifyFileInLog(v, "noicompile/m-n-test/client/target/m-n-test-1.0-SNAPSHOT-javafx.zip");
         
         File nbactions = new File(created, "nbactions.xml");
         assertTrue(nbactions.isFile(), "Actions file is in there");
@@ -261,6 +269,12 @@ public class VerifyArchetypeIT {
         assertTrue(created.isDirectory(), "Project created");
         assertTrue(new File(created, "pom.xml").isFile(), "Pom file is in there");
 
+        File index = new File(new File(new File(new File(new File(
+            created, "src"), "main"), "webapp"), "pages"), "index.html");
+        assertTrue(index.exists(), "HTML page found " + index);
+        File bin = new File(index.getParentFile(), "index.bin");
+        writeBinary(bin);
+        
         File and = new File(new File(dir, "d-l-test"), "android-test");
         assertTrue(and.isDirectory(), "Project created");
         assertTrue(new File(and, "pom.xml").isFile(), "Pom file is in there");
@@ -299,6 +313,9 @@ public class VerifyArchetypeIT {
         
         JarFile jf = new JarFile(apk);
         assertNotNull(jf.getEntry("assets/pages/index.html"), "index.html is included in " + apk);
+        ZipEntry indexBin = jf.getEntry("assets/pages/index.bin");
+        assertNotNull(indexBin, "binary file found in " + apk);
+        assertBinary(jf.getInputStream(indexBin));
         jf.close();
     }
 
@@ -323,7 +340,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("verify");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("wandroidcmp/w-d-test/client/target/w-d-test-1.0-SNAPSHOT-javafx.zip");
+        verifyFileInLog(v, "wandroidcmp/w-d-test/client/target/w-d-test-1.0-SNAPSHOT-javafx.zip");
     }
 
     @Test
@@ -334,14 +351,17 @@ public class VerifyArchetypeIT {
         File created = new File(new File(dir, "b-p-test"), "client");
         assertTrue(created.isDirectory(), "Project created");
         assertTrue(new File(created, "pom.xml").isFile(), "Pom file is in there");
-        
-        File web = new File(new File(dir, "b-p-test"), "test-web");
-        assertTrue(web.isDirectory(), "Project created");
-        assertTrue(new File(web, "pom.xml").isFile(), "Pom file is in there");
-        
+
         File main = new File(new File(created, "src"), "main");
         File pages = new File(new File(main, "webapp"), "pages");
         File index = new File(pages, "index.html");
+        assertTrue(index.exists(), "HTML page found " + index);
+        File bin = new File(index.getParentFile(), "index.bin");
+        writeBinary(bin);
+                
+        File web = new File(new File(dir, "b-p-test"), "test-web");
+        assertTrue(web.isDirectory(), "Project created");
+        assertTrue(new File(web, "pom.xml").isFile(), "Pom file is in there");
         
         String indexContent = Files.readFile(index);
         assertTrue(indexContent.contains("src=\"bck2brwsr.js\""), "There should be bck2brwsr.js reference in " + index);
@@ -358,7 +378,7 @@ public class VerifyArchetypeIT {
         v.executeGoal("package");
         
         v.verifyErrorFreeLog();
-        v.verifyTextInLog("b2bcmp/b-p-test/test-web/target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
+        verifyFileInLog(v, "b2bcmp/b-p-test/test-web/target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
         
         v.assertFileNotPresent("target/res/drawable-hdpi/ic_launcher.png");
         v.assertFileNotPresent("target/res/drawable-mdpi/ic_launcher.png");
@@ -370,6 +390,10 @@ public class VerifyArchetypeIT {
         v.assertFilePresent("target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
         v.assertFilePresent("target/b-p-test.js");
         v.assertFilePresent("target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr/public_html/index.html");
+        v.assertFilePresent("target/b-p-test-web-1.0-SNAPSHOT-bck2brwsr/public_html/index.bin");
+        File indexBin = new File(new File(new File(new File(web, "target"), "b-p-test-web-1.0-SNAPSHOT-bck2brwsr"), "public_html"), "index.bin");
+        assertTrue(indexBin.exists(), "index.bin really exists");
+        assertBinary(new FileInputStream(indexBin));
 
         File nbactions = new File(web, "nbactions.xml");
         assertTrue(nbactions.isFile(), "Actions file is in there");
@@ -417,7 +441,7 @@ public class VerifyArchetypeIT {
             v.executeGoal("package");
 
             v.verifyErrorFreeLog();
-            v.verifyTextInLog("BandN/b-n-test/for-web/target/b-n-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
+            verifyFileInLog(v, "BandN/b-n-test/for-web/target/b-n-test-web-1.0-SNAPSHOT-bck2brwsr.zip");
 
             v.assertFileNotPresent("target/res/drawable-hdpi/ic_launcher.png");
             v.assertFileNotPresent("target/res/drawable-mdpi/ic_launcher.png");
@@ -563,4 +587,20 @@ public class VerifyArchetypeIT {
         Document dom = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(u.openStream());
         return xp.evaluate(dom);
     }    
+    
+    static void writeBinary(File out) throws IOException {
+        FileOutputStream fos = new FileOutputStream(out);
+        for (int i = 0; i < 256; i++) {
+            fos.write(i);
+        }
+        fos.close();
+    }
+    
+    static void assertBinary(InputStream is) throws IOException {
+        for (int i = 0; i < 256; i++) {
+            int b = is.read();
+            assertEquals((byte)b, (byte)i, i + "th byte of the stream should be equal");
+        }
+        is.close();
+    }
 }
diff --git a/ko-archetype/src/main/java/com/dukescript/archetype/crud/package-info.java b/ko-archetype/src/main/java/com/dukescript/archetype/crud/package-info.java
deleted file mode 100644
index 68bf7e8..0000000
--- a/ko-archetype/src/main/java/com/dukescript/archetype/crud/package-info.java
+++ /dev/null
@@ -1,24 +0,0 @@
-/**
- * The MIT License (MIT)
- *
- * Copyright (C) 2015 Anton Epple <toni.epple@eppleton.de>
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
-package com.dukescript.archetype.crud;
diff --git a/ko-archetype/src/main/java/com/dukescript/archetype/ko/package-info.java b/ko-archetype/src/main/java/com/dukescript/archetype/ko/package-info.java
new file mode 100644
index 0000000..571db9e
--- /dev/null
+++ b/ko-archetype/src/main/java/com/dukescript/archetype/ko/package-info.java
@@ -0,0 +1,24 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (C) 2015 Anton Epple <toni.epple@eppleton.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.dukescript.archetype.ko;
diff --git a/ko-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml b/ko-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml
index f42dd74..12773f6 100644
--- a/ko-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml
+++ b/ko-archetype/src/main/resources/archetype-resources/__androidpath__/nbactions.xml
@@ -22,6 +22,8 @@
         </goals>
         <properties>
             <skipTests>true</skipTests>
+            <android.run.debug>${jpda.attach.port}</android.run.debug>
+            <jpda.attach>true</jpda.attach>
         </properties>
     </action>
     <action>
diff --git a/ko-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml b/ko-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml
index 56a713a..5df7721 100644
--- a/ko-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml
+++ b/ko-archetype/src/main/resources/archetype-resources/__androidpath__/pom.xml
@@ -18,7 +18,7 @@
 
     <properties>
         <platform.version>4.1.1.4</platform.version>
-        <android.plugin.version>4.1.0</android.plugin.version>
+        <android.plugin.version>4.2.0</android.plugin.version>
         <android.platform.version>19</android.platform.version>
         <debug>false</debug>
     </properties>
diff --git a/ko-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml b/ko-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml
index 2ff682b..a445c5d 100644
--- a/ko-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml
+++ b/ko-archetype/src/main/resources/archetype-resources/__iospath__/nbactions.xml
@@ -18,6 +18,9 @@
         </goals>
         <properties>
             <skipTests>true</skipTests>
+            <robovm.debug>clientmode</robovm.debug>
+            <robovm.debugPort>${jpda.address}</robovm.debugPort>
+            <jpda.listen>true</jpda.listen>
         </properties>
     </action>
     <action>
diff --git a/ko-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml b/ko-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml
index 4e8b08b..bbe7cc6 100644
--- a/ko-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml
+++ b/ko-archetype/src/main/resources/archetype-resources/__iospath__/pom.xml
@@ -145,16 +145,13 @@
             <plugin>
                 <groupId>org.robovm</groupId>
                 <artifactId>robovm-maven-plugin</artifactId>
-                <version>1.0.0-beta-03</version>
+                <version>${robovm.version}</version>
                 <configuration>
                     <config>
                         <mainClass>\${project.mainclass}</mainClass>
                         <forceLinkClasses>
                             <pattern>java.util.logging.ConsoleHandler</pattern>
                             <pattern>java.util.logging.SimpleFormatter</pattern>
-                            <pattern>org.netbeans.html.ko4j.KO4J</pattern>
-                            <pattern>org.netbeans.html.sound.impl.BrowserAudioEnv</pattern>
-                            <pattern>com.dukescript.presenters.iOS</pattern>
                         </forceLinkClasses>
                         <executableName>\${artifactId}</executableName>
                         <resources>
diff --git a/ko-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml b/ko-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml
index 9114753..2427596 100644
--- a/ko-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml
+++ b/ko-archetype/src/main/resources/archetype-resources/__webpath__/src/main/assembly/bck2brwsr.xml
@@ -37,7 +37,6 @@
       <fileSet>
           <directory>target/web/pages</directory>
           <outputDirectory>/</outputDirectory>
-          <filtered>true</filtered>
       </fileSet>
         <fileSet>
             <directory>${project.build.directory}</directory>
diff --git a/ko-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml b/ko-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml
index 871719b..dd4b2f3 100644
--- a/ko-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml
+++ b/ko-archetype/src/main/resources/archetype-resources/client/src/main/assembly/webpages.xml
@@ -36,22 +36,7 @@
       <fileSet>
           <directory>src/main/webapp/pages</directory>
           <outputDirectory>/</outputDirectory>
-          <filtered>true</filtered>
-          <excludes>
-              <exclude>**/*.png</exclude>
-              <exclude>**/*.jpg</exclude>
-              <exclude>**/*.gif</exclude>
-          </excludes>
-      </fileSet>
-      <fileSet>
-          <directory>src/main/webapp/pages</directory>
-          <outputDirectory>/</outputDirectory>
           <filtered>false</filtered>
-          <includes>
-              <include>**/*.png</include>
-              <include>**/*.jpg</include>
-              <include>**/*.gif</include>
-          </includes>
       </fileSet>
   </fileSets>
 </assembly>
diff --git a/pom.xml b/pom.xml
index 928ec02..f7446b8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -36,8 +36,9 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <net.java.html.version>1.1</net.java.html.version>
         <nb.html.version>1.0</nb.html.version>
-        <presenters.version>0.7</presenters.version>
-        <bck2brwsr.version>0.13</bck2brwsr.version>
+        <presenters.version>0.8.1</presenters.version>
+        <robovm.version>1.2.0</robovm.version>
+        <bck2brwsr.version>0.14</bck2brwsr.version>
         <jersey.version>2.13</jersey.version>
         <license>COPYING</license>
         <netbeans.compile.on.save>none</netbeans.compile.on.save>
